src/main/java/model/FruitTransaction.java

```diff
@@ -0,0 +1,55 @@
+package model;
+
+public class FruitTransaction {
+    private Operation operation;
+    private String fruit;
+    private int quantity;
+
+    public Operation getOperation() {
+        return operation;
+    }
+
+    public void setOperation(String op) {
+        this.operation = switch (op) {
+            case "b" -> Operation.BALANCE;
+            case "s" -> Operation.SUPPLY;
+            case "p" -> Operation.PURCHASE;
+            case "r" -> Operation.RETURN;
+            default -> throw new IllegalArgumentException(
+                    "There is no Enum element, corresponding to the argument " + op);
+        };
+    }
+
+    public String getFruit() {
+        return fruit;
+    }
+
+    public void setFruit(String fruit) {
+        this.fruit = fruit;
+    }
+
+    public int getQuantity() {
+        return quantity;
+    }
+
+    public void setQuantity(int quantity) {
+        this.quantity = quantity;
+    }
+
+    public enum Operation {
```

Comment:
The `Operation` enum is declared inside the `Transaction` class. This violates the Single Responsibility Principle (SRP) because the `Transaction` class is now responsible for both representing a transaction and representing an operation. It would be better to move the `Operation` enum to a separate file.